raysect/optical/observer/imaging/fieldofview.pyx

```diff
@@ -0,0 +1,224 @@
+    """
+    cdef:
+        double _sensitivity, azimuth_delta, altitude_delta
+        np.ndarray _azimuth, _altitude
+        tuple _fov
+        RectangleSampler3D point_sampler
+
+    def __init__(self, pixels, fov=(90., 90.), sensitivity=1.0, frame_sampler=None,
+                 pipelines=None, parent=None, transform=None, name=None):
+
+        # defaults to an adaptively sampled RGB pipeline
+        if not pipelines and not frame_sampler:
+            rgb = RGBPipeline2D()
+            pipelines = [rgb]
+            frame_sampler = RGBAdaptiveSampler2D(rgb)
+        else:
+            pipelines = pipelines or [RGBPipeline2D()]
+            frame_sampler = frame_sampler or FullFrameSampler2D()
+
+        self._fov = (90., 90.)
+        super().__init__(pixels, frame_sampler, pipelines, parent=parent, transform=transform, name=name)
+
+        # note that the fov property triggers a call to _update_image_geometry()
+        self.fov = fov
+        self.sensitivity = sensitivity
+
+    @property
+    def fov(self):
+        return self._fov
+
+    @fov.setter
+    def fov(self, value):
+        value = tuple(value)
+        if len(value) != 2:
+            raise ValueError("Fov must be a 2 element tuple defining the field-of-view in horizontal and vertical directions")
+        if value[0] <= 0 or value[0] >= 180 or value[1] <= 0 or value[1] >= 180:
+            raise ValueError("The field-of-view angle must lie in the range (0, 180).")
+        self._fov = value
+        self._update_image_geometry()
+
+    @property
+    def pixels(self):
+        return self._pixels
+
+    @pixels.setter
+    def pixels(self, value):
+        pixels = tuple(value)
+        if len(pixels) != 2:
+            raise ValueError("Pixels must be a 2 element tuple defining the x and y resolution.")
+        x, y = pixels
+        if x <= 0:
+            raise ValueError("Number of x pixels must be greater than 0.")
+        if y <= 0:
+            raise ValueError("Number of y pixels must be greater than 0.")
+        self._pixels = pixels
+        self._update_image_geometry()
+
+    @property
+    def sensitivity(self):
+        return self._sensitivity
+
+    @sensitivity.setter
+    def sensitivity(self, value):
+        if value <= 0:
+            raise ValueError("Sensitivity must be greater than zero.")
+        self._sensitivity = value
+
+    @property
+    def azimuth(self):
+        return self._azimuth
+
+    @property
+    def altitude(self):
+        return self._altitude
+
+    @property
+    def angle_h(self):
+        return self._azimuth
+
+    @property
+    def angle_v(self):
+        return self._altitude
+
+    cpdef np.ndarray solid_angle(self):
+        """
+        Returns solid angles of collection (in steradians) for all pixels.
+        """
+
+        cdef:
+            np.ndarray solid_angle1d
+
+        solid_angle1d = (pi / 180.) * self.azimuth_delta * (np.sin((pi / 180.) * (self._altitude + 0.5 * self.altitude_delta)) -
```

Comment:
This doesn't look right to me. d(SA) = sin(theta) d(theta) d(phi). The surface integral over an element spanning [theta0, theta1] and [phi0, phi1] should include a cos term.... I hope this is not an approximation.

Reply:
Here, the altitude angle is measured from the equatorial plane, not from the pole (Z-axis is also lies in the equatorial plane, it easier when dealing with a field of view), so d(SA) = cos(theta) d(theta) d(phi).

Reply:
Not sure I followed the description. The raysect standard is for the camera to look along the +ve z-axis, with the "up" direction being +ve y-axis. The altitude here should be the equivalent of the pitch in math.transform.rotate(), while the azimuth should be the yaw. (0, 0) degrees being aligned with the z-axis. Is this what you mean? I suspect it is.

Reply:
If it is then it should be cos(theta) d(theta) d(phi) as you say. As the circumference decreases as the angle altitude increases.

Reply:
The camera looks along +ve z-axis and the "up" direction is +ve y-axis just like in all other Raysect cameras. However, altitude and azimuth angles are calculated like in [horizontal coordinate system](https://en.wikipedia.org/wiki/Horizontal_coordinate_system), so the direction of z-axis is not the azimuth point but the north point (see the scheme). For the FoV with angles of view (phi, theta), the azimuth angle goes from -phi/2 to phi/2 and the altitude angle goes from -theta/2 to theta/2. The total solid angle observed is 2 * phi * sin(theta/2).

Reply:
Ah we are describing the same thing, great.